ngi_reports/utils/entities.py

```diff
@@ -43,18 +57,258 @@ def __init__(self):
 class Flowcell:
     """Flowcell class"""
 
-    def __init__(self):
-        self.date = ""
+    def __init__(self, fc, ngi_name, db_connection):
+        self.fc = fc
+        self.project_name = ngi_name
+        self.db_connection = db_connection
+        self.name = self.fc.get("name", "")
+        self.run_name = self.fc.get("run_name", "")
+        self.date = self.fc.get("date", "")
+        self.fc_details = self.db_connection.get_entry(self.run_name)
+
         self.lanes = OrderedDict()
-        self.name = ""
-        self.run_name = ""
-        self.run_setup = []
-        self.seq_meth = ""
-        self.type = ""
-        self.run_params = {}
-        self.chemistry = {}
-        self.casava = None
-        self.seq_software = {}
+        self.fc_sample_qvalues = defaultdict(dict)
+
+    def populate_illumina_flowcell(self, log, **kwargs):
+        fc_instrument = self.fc_details.get("RunInfo", {}).get("Instrument", "")
+        if "-" in self.name:
+            self.type = "MiSeq"
+            fc_runparameters = self.fc_details.get("RunParameters", {})
+        elif fc_instrument.startswith("A"):
+            self.type = "NovaSeq6000"
+            fc_runparameters = self.fc_details.get("RunParameters", {})
+        elif fc_instrument.startswith("LH"):
+            self.type = "NovaSeqXPlus"
+            fc_runparameters = self.fc_details.get("RunParameters", {})
+        elif fc_instrument.startswith("NS"):
+            self.type = "NextSeq500"
+            fc_runparameters = self.fc_details.get("RunParameters", {})
+        elif fc_instrument.startswith("VH"):
+            self.type = "NextSeq2000"
+            fc_runparameters = self.fc_details.get("RunParameters", {})
+
+        self.run_setup = self.fc_details.get("RunInfo").get("Reads")
+
+        if self.type == "NovaSeq6000":
```

Comment:
Move chemistry up to where type is defined?

Reply:
Done